share/functions/funcsave.fish

```diff
@@ -8,20 +8,26 @@ function funcsave --description "Save the current definition of all specified fu
         return 0
     end
 
+	if set -q _flag_directory
+		set funcdir $_flag_directory
+	else
+		set funcdir $__fish_config_dir/functions
+	end
+
     if not set -q argv[1]
         printf (_ "%ls: Expected at least %d args, got only %d\n") funcsave 1 0
         return 1
     end
 
-    if not mkdir -p $__fish_config_dir/functions
-        printf (_ "%s: Could not create configuration directory\n") funcsave
+    if not mkdir -p $funcdir
+        printf (_ "%s: %s: Could not create configuration directory\n") funcsave $funcdir
```

Comment:
This reads as

    "funcsave: /home/alfa/.config/fish/test/functions: Could not create configuration directory"

I think it's better as 

    "funcsave: Could not create directory '/home/alfa/.config/fish/test/functions'"

Reply:
Fixed